### PR TITLE
fix x2y2 airdrop query

### PR DESCRIPTION
**Pull request by ptPierre**
### Update!
Please build spells in the proper [subproject](../dbt_subprojects/) directory. For more information, please see the main [readme](../README.md), which also links to a GH discussion with the option to ask questions.

### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [ ] Adding to existing spell lineage
- [X] Bug fix

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** the query tracked token transfers and not the airdrop claims
- **Steps to reproduce:** run the current query and sum up the amount_original field, compare it with max supply
- **Implementation details:** The contract was changed to the airdrop contract

---

Thank you for your contribution!

### CODE REVIEW

**Comment by ptPierre** [comment]
I have read the CLA Document and I hereby sign the CLA

File: dbt_subprojects/daily_spellbook/models/_sector/airdrops/ethereum/projects/x2y2_ethereum_airdrop_claims.sql [hunk 1]
```diff
@@ -18,7 +18,7 @@
 WITH early_price AS (
     SELECT MIN(minute) AS minute
     , MIN_BY(price, minute) AS price
-    FROM {{ source('prices', 'usd') }}
+    FROM prices.usd
```
Comment (Hosuke) [inline dbt_subprojects/daily_spellbook/models/_sector/airdrops/ethereum/projects/x2y2_ethereum_airdrop_claims.sql:21]: We need to use the {{}} format for sources and refs in the spell.
```suggestion
    FROM {{ source('prices', 'usd') }}
```

File: dbt_subprojects/daily_spellbook/models/_sector/airdrops/ethereum/projects/x2y2_ethereum_airdrop_claims.sql [hunk 2]
```diff
@@ -28,20 +28,19 @@ SELECT 'ethereum' AS blockchain
...
+, CASE WHEN t.evt_block_time >= (SELECT minute FROM early_price) THEN CAST(pu.price*t.amount/POWER(10, 18) AS double)
+    ELSE CAST((SELECT price FROM early_price)*t.amount/POWER(10, 18) AS double)
     END AS amount_usd
 , {{xtyt_token_address}} AS token_address
 , 'X2Y2' AS token_symbol
 , t.evt_index
-FROM {{ source('erc20_ethereum', 'evt_transfer') }} t
-LEFT JOIN {{ source('prices','usd_forward_fill') }} pu ON pu.blockchain = 'ethereum'
+FROM x2y2_ethereum.X2Y2Drop_evt_Airdrop t
+LEFT JOIN prices.usd_forward_fill pu ON pu.blockchain = 'ethereum'
```
Comment (Hosuke) [inline dbt_subprojects/daily_spellbook/models/_sector/airdrops/ethereum/projects/x2y2_ethereum_airdrop_claims.sql:43]:
```suggestion
LEFT JOIN {{ source('prices', 'usd_forward_fill') }} pu ON pu.blockchain = 'ethereum'
```

File: dbt_subprojects/daily_spellbook/models/_sector/airdrops/ethereum/projects/x2y2_ethereum_airdrop_claims.sql [hunk 3]
```diff
@@ -28,20 +28,19 @@ SELECT 'ethereum' AS blockchain
...
+, CAST(t.amount/POWER(10, 18) AS double) AS amount_original
+, CASE WHEN t.evt_block_time >= (SELECT minute FROM early_price) THEN CAST(pu.price*t.amount/POWER(10, 18) AS double)
+    ELSE CAST((SELECT price FROM early_price)*t.amount/POWER(10, 18) AS double)
     END AS amount_usd
 , {{xtyt_token_address}} AS token_address
 , 'X2Y2' AS token_symbol
 , t.evt_index
-FROM {{ source('erc20_ethereum', 'evt_transfer') }} t
-LEFT JOIN {{ source('prices','usd_forward_fill') }} pu ON pu.blockchain = 'ethereum'
+FROM x2y2_ethereum.X2Y2Drop_evt_Airdrop t
```
Comment (Hosuke) [inline dbt_subprojects/daily_spellbook/models/_sector/airdrops/ethereum/projects/x2y2_ethereum_airdrop_claims.sql]:
```suggestion
FROM {{ source('x2y2_ethereum', 'X2Y2Drop_evt_Airdrop') }} t
```

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @ptPierre 

[Sample x2y2_ethereum.airdrop_claims](https://dune.com/queries/3986015)

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**